### PR TITLE
Fix report issues

### PR DESCRIPTION
**Pull request by eshilov**

### CODE REVIEW

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import java.util.Date;
+import java.util.UUID;
+
+import static com.johncnstn.auth.security.DomainGrantedAuthority.USER;
+import static io.jsonwebtoken.SignatureAlgorithm.HS512;
+import static io.jsonwebtoken.io.Decoders.BASE64;
+import static java.lang.System.currentTimeMillis;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+public class TokensProviderTest {
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: Why is it not extending one of abstract classes for tests?

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java [hunk 2]
```diff
@@ -20,11 +20,11 @@
     @Override
     @Transactional(readOnly = true)
     @SuppressWarnings("")
```
Comment (JohnCnstn) [inline src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java]: Remove this @SuppressWarnings(""), please

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java [hunk 3]
```diff
@@ -20,11 +20,11 @@
...
-    public DomainUserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+    public DomainUserDetails loadUserByUsername(String username) {
```
Comment (JohnCnstn) [inline src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:22]: This throws UsernameNotFoundException have to be returned, thought.

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java [hunk 4]
```diff
@@ -20,11 +20,11 @@
...
         return loadUserByEmail(username);
     }
 
-    private DomainUserDetails loadUserByEmail(String email) throws UsernameNotFoundException {
+    private DomainUserDetails loadUserByEmail(String email) {
```
Comment (JohnCnstn) [inline src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:26]: This throws UsernameNotFoundException have to be returned, thought.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package com.johncnstn.auth.constraints;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PasswordValidatorTest {
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Why is it not extending one of abstract classes for tests?

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Test
+    public void testEmpty() {
+        testPassword("", false);
+    }
+
+    @Test
+    public void testWithWhitespace() {
+        testPassword("Password 1", false);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: For fake data use java faker.
View utils class for this purposes.

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    @Mock
+    private UserRepository userRepository;
+
+    @Test
+    public void testLoadExistingUser() {
+        // GIVEN
+        var userDetailsService = new DomainUserDetailsService(userRepository);
+        var email = "test@mail.com";
+        var userEntity = new UserEntity(UUID.randomUUID(), email, "demo123", UserRoleEntity.USER);
+        Mockito.when(userRepository.findByEmail(email)).thenReturn(userEntity);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Import this Mockito, please.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package com.johncnstn.auth.constraints;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PasswordValidatorTest {
+
+    @Test
+    public void testNull() {
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Remove test from test methods everywhere.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Test
+    public void testFine() {
+        testPassword("Password1", true);
+    }
+
+    @Test
+    private void testPassword(String password, boolean expected) {
+        // WHEN
+        var result = PasswordValidator.isValid(password);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: PasswordValidator import, please.

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+            it.assertThat(userDetails.getAuthorities()).containsExactly(DomainGrantedAuthority.USER);
+        });
+    }
+
+    @Test
+    public void testLoadNotExistingUser() {
+        // GIVEN
+        var userDetailsService = new DomainUserDetailsService(userRepository);
+        var email = "test@mail.com";
+        Mockito.when(userRepository.findByEmail(email)).thenReturn(null);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Import Mockito.

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Test
+    public void testLoadExistingUser() {
+        // GIVEN
+        var userDetailsService = new DomainUserDetailsService(userRepository);
+        var email = "test@mail.com";
+        var userEntity = new UserEntity(UUID.randomUUID(), email, "demo123", UserRoleEntity.USER);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Use utils class.

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+public class DomainUserDetailsServiceTest extends AbstractUnitTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Test
+    public void testLoadExistingUser() {
+        // GIVEN
+        var userDetailsService = new DomainUserDetailsService(userRepository);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Don't create beans by yourself, it's unit test, other classes have to be mocked, that's it.

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java (hunk 1 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: Why should it be? Do all our unit test really need this?
`AbstractUnitTest` class has quite a lot of overhead including
```
@AutoConfigureMockMvc
@ExtendWith(SpringExtension.class)
-----
@MockBean
protected HttpClient httpClient;
@MockBean
protected HttpResponse<Object> httpResponse;
```

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java (hunk 3 above)
Comment (eshilov) [reply src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:22]: PMD is complaining about declaring unchecked exceptions to be thrown

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java (hunk 1 above)
Comment (JohnCnstn) [reply src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: 1) This parent class is only for pre-config.
@AutoConfigureMockMvc
@ExtendWith(SpringExtension.class) - for example.
Also, mocks will be helpful not in all child classes, but in many.
2) It's much easier to orient in the code.
All tests have to be extended from one of two classes.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 6 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Okay
Could you please point me to a faker method which will create a string which is a valid password with the only exception that it contains a whitespace?

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java (hunk 3 above)
Comment (JohnCnstn) [reply src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:22]: Yeah
So, you have to find how to write @SupressWarning annotation and pass the pmd check.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 9 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: I feel that here, we should not do a static import of PasswordValidator.isValid as it makes less clear about what we are actually testing.

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java (hunk 1 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: Done

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java (hunk 2 above)
Comment (eshilov) [reply src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java]: Done

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java (hunk 3 above)
Comment (eshilov) [reply src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:22]: Left it as it is, as per offline discussion.

File: src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java (hunk 4 above)
Comment (eshilov) [reply src/main/java/com/johncnstn/auth/security/DomainUserDetailsService.java:26]: Left it as it is, as per offline discussion.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 5 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Done

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java (hunk 7 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Done

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 8 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Removed test from all tests in this file.

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java (hunk 10 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Done

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java (hunk 11 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Done

File: src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java (hunk 12 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/security/DomainUserDetailsServiceTest.java]: Used @InjectMocks for test objects initialization.

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+        // WHEN
+        JwtTokens tokens = tokensProvider.refreshTokens(accessToken, refreshToken);
+
+        // THEN
+        assertSoftly(it -> {
+            it.assertThat(tokens).isNotNull();
+            it.assertThat(tokens.getIssuedAt()).isGreaterThan(0);
+            it.assertThat(tokens.getIssuedAt()).isLessThanOrEqualTo(currentTimeMillis());
+            it.assertThat(tokens.getAccessToken()).isNotNull();
+            it.assertThat(tokens.getAccessExpiresIn()).isEqualTo(tokens.getIssuedAt() + 86400000L);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: 86400000L this should be a constant
Also, use time api that were introduced in java 8

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+        var accessToken = createToken(
+                userDetails,
+                issuedAt,
+                JwtTokenType.ACCESS,
+                new Date(2020, 5, 10));
+
+        var refreshToken = createToken(
+                userDetails,
+                issuedAt,
+                JwtTokenType.REFRESH,
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: JwtTokenType import

File: src/test/java/com/johncnstn/auth/security/TokensProviderTest.java [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    public static void setUp() {
+        appProperties = new AppProperties();
+        tokensProvider = new TokensProvider(appProperties);
+        tokensProvider.init();
+    }
+
+    @Test
+    public void testCreateTokens() {
+        // GIVEN
+        var principal = new DomainUserDetails(UUID.randomUUID(), USER);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/security/TokensProviderTest.java]: UUID import

File: src/test/java/com/johncnstn/auth/service/AuthServiceTest.java [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+        // GIVEN
+        var rawUser = new User();
+        rawUser.setEmail("test@mail.com");
+        rawUser.setPassword("demo123");
+
+        var entityToReturn = new UserEntity(
+                UUID.randomUUID(),
+                rawUser.getEmail(),
+                "xyz",
+                UserRoleEntity.USER);
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/service/AuthServiceTest.java]: UserRoleEntity import

File: src/test/java/com/johncnstn/auth/util/TestUtil.kt [hunk 17]
```diff
@@ -1,14 +1,21 @@
 package com.johncnstn.auth.util
```
Comment (JohnCnstn) [inline src/test/java/com/johncnstn/auth/util/TestUtil.kt]: Rewrite this class in java

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 9 above)
Comment (JohnCnstn) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: We don't need info from where we are using this method.
Just isValid and password as a variable would be sufficient.

Comment (eshilov) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: Here, we actually need this info. I agree that it's not needed when we are using this method in code. In code, we just need to state that we want to validate a password, not depending on where we are taking the validation routine, abstracting away the actual validation logic owner. In a test method, we want to say that we are testing a concrete method of a concrete class. If you still feel that we have to omit the class here, I will do it as it is a minor thing.

File: src/test/java/com/johncnstn/auth/service/AuthServiceTest.java (hunk 16 above)
Comment (eshilov) [reply src/test/java/com/johncnstn/auth/service/AuthServiceTest.java]: There are three constants with the name USER in this class.

File: src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java (hunk 9 above)
Comment (JohnCnstn) [reply src/test/java/com/johncnstn/auth/constraints/PasswordValidatorTest.java]: As we discussed earlier, let's name methods like isPasswordValid for static methods.
If it's @Component - then just isValid.

**Pull request merged**